### ISSUE TITLE
Disabling IPv6 Annotation at Deployment Spec

### ISSUE THREAD

**Issue by ravish-kumar-maersk** [kumahq/kuma#9121]
### What happened?

We are using Kuma 2.4.1 in multi-zone deployment. 
Global CP is hosted in AKS.

A managed k8s is created using Azure VMs and added as a zone. It only uses IPv4. 

So per the [documentation](https://kuma.io/docs/2.4.x/production/dp-config/ipv6/#disabling-ipv6), need to disable IPv6 from DP config.

We are applying the below annotation at the "Deployment" level but it's **NOT** getting applied to Pod. 
`kuma.io/transparent-proxying-inbound-v6-port: "0"`

Need to add the annotation specifically at Pod Spec.  

Is this expected behavior?? 

Deployment Spec -- 

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: app2-deployment
  namespace: kuma-demo
  labels:
    app: app2
spec:
  replicas: 1
  selector:
    matchLabels:
      app: app2
  template:
    metadata:
      annotations:
        kuma.io/transparent-proxying-inbound-v6-port: "0"
      labels:
        app: app2
        kuma.io/sidecar-injection: enabled
    spec:
      containers:
      - name: demo
        image: xxxxxxxx.azurecr.io/demo
```

**Comment by slonka** [comment]
Reproduced this locally on k3d:

```
$ kubectl get deployments.apps -n kuma-demo demo-app -o yaml | yq '.spec.template.metadata.annotations'
kuma.io/transparent-proxying-inbound-v6-port: "0"
my.custom.annotations.com: "1"
```

```
kubectl get pods -n kuma-demo demo-app-6d9ff469f8-hq8k2 -o yaml | yq '.metadata.annotations'
kubectl.kubernetes.io/default-container: demo-app
kuma.io/builtin-dns: enabled
kuma.io/builtin-dns-port: "15053"
kuma.io/envoy-admin-port: "9901"
kuma.io/mesh: default
kuma.io/sidecar-injected: "true"
kuma.io/sidecar-uid: "5678"
kuma.io/transparent-proxying: enabled
kuma.io/transparent-proxying-ebpf: disabled
kuma.io/transparent-proxying-inbound-port: "15006"
kuma.io/transparent-proxying-inbound-v6-port: "15010"
kuma.io/transparent-proxying-outbound-port: "15001"
kuma.io/virtual-probes: enabled
kuma.io/virtual-probes-port: "9000"
my.custom.annotations.com: "1"
```

**Comment by lukidzi** [comment]
> Triage: It should be fixed in the new release 2.9.x https://github.com/kumahq/kuma/pull/10906 You can use use only ipv4 by setting `kuma.io/transparent-proxying-ip-family-mode: ipv4`
